extension/intents/navigation/navigation.js

```diff
@@ -66,15 +67,37 @@ intentRunner.registerIntent({
 intentRunner.registerIntent({
   name: "navigation.bangSearch",
   async run(context) {
-    const service = context.slots.service || context.parameters.service;
-    const myurl = await searching.ddgBangSearchUrl(
-      context.slots.query,
-      service
-    );
-    context.addTelemetryServiceName(
-      `ddg:${serviceList.ddgBangServiceName(service)}`
-    );
-    await context.createTab({ url: myurl });
+    const serviceType =
+      context.slots.service || context.parameters.service
+        ? "mentioned"
+        : "notMentioned";
+    let myurl;
+    const service =
+      serviceType === "mentioned"
+        ? context.slots.service || context.parameters.service
+        : await serviceList.detectServiceFromActiveTab(metadata.search);
+    if (service) {
+      myurl = await searching.ddgBangSearchUrl(context.slots.query, service);
+      context.addTelemetryServiceName(
+        `ddg:${serviceList.ddgBangServiceName(service)}`
+      );
+      if (serviceType === "mentioned") {
+        await context.createTab({ url: myurl });
+      } else {
+        await browser.tabs.update({ url: myurl });
+      }
+    } else {
+      myurl = new URL((await browserUtil.activeTab()).url);
+      const tab = await context.activeTab();
+      await browser.search.search({
+        query:
+          myurl.origin !== "null" && myurl.href !== "about:blank"
+            ? context.slots.query + ` site:${myurl.origin}`
+            : context.slots.query,
+        tabId: tab.id,
+      });
+    }
```

Comment:
This else part displays search results in default search engine with search bar displaying:
“[query] site:https://www.example.com”
For example, I opened yahoo.com and gave command "search sports here", above code gives results like this: 
https://duckduckgo.com/?t=ffsb&q=sports+site%3Ahttps%3A%2F%2Fwww.yahoo.com&ia=web

Instead of displaying the search results from the default search engine, I want to perform  search using the relevant service and display results from that site. To serve the purpose of this issue (which was: **search here**)
I want to change this else part to something like this: 
```
const tabOrigin = new URL((await browserUtil.activeTab()).url).origin;
const startIndex = tabOrigin.indexOf(".");  
const endIndex = tabOrigin.indexOf(".", startIndex+1);  
const unKnownService = tabOrigin.substring(startIndex+1, endIndex); 
myurl = await searching.searchUrlUnknownService(context.slots.query, unKnownService); 
await browser.tabs.update({ url: myurl });
```
Keeping searchUrlUnknownService() method similar to ddgBangSearchUrl(), except it does not check metadata for service bang, it is given the service name from the active tab’s url and it generates the search url with given service name.
Repeating the same example:  I opened yahoo.com and gave command "search sports here". This code gives results which are more desirable, like this: 
https://search.yahoo.com/search?p=sports

Reply:
Now what I want to do is to search https://duckduckgo.com/bang to bring in the bang from ddg.
I recently noticed from #1475, that we can do searches in a hidden tab and use the results. 
I am able to redirect to ddgbang searches to get the right bang. I believe we can bring bangs from ddg and redirect to any website the user wants. 
@ianb can you guide me in the right direction and code, How can I perform any task in hidden tab and bring in the search results from there?